assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -254,7 +255,11 @@ export default Component.extend(TextareaTextManipulation, ComposerUploadUppy, {
   didReceiveAttrs() {
     this._super(...arguments);
 
-    if (!this.editingMessage && this.draft) {
+    if (
+      !this.editingMessage &&
+      this.draft &&
+      this.chatChannel.canModifyMessages(this.currentUser)
+    ) {
```

Comment:
Something else to think of is if we delete drafts for channels that have been closed?

Reply:
Good question -- I don't think we do anything to drafts for topics that are closed, so IMO we should leave these too